api/routes/router.go

```diff
@@ -11,6 +11,8 @@ func SetupRoutes(engine *gin.Engine) {
 	engine.POST("/signup", controllers.HandleSignUp)
 	engine.POST("/code_validation", controllers.HandleCodeValidation)
 	engine.POST("/new_code", controllers.HandleNewCodeValidation)
+	engine.POST("/reset_password", controllers.HandleCodeResetPassword)
+	engine.PUT("/reset_password", controllers.HandleResetPassword)
```

Comment:
What do you think about: 

`/user/password/code` for the POST route and
`/user/password` for the PUT route.